bvm/ballerina-core/src/main/java/org/ballerinalang/util/observability/ObserverContext.java

```diff
@@ -120,6 +122,14 @@ void setStarted() {
         this.started = true;
     }
 
+    public boolean isFinished() {
+        return finished;
+    }
+
+    public void setFinished() {
```

Comment:
This should be package-private. See also setStarted() and setServer()

Reply:
Since setFinished() is used from the observe package as well it has been made public